samtranslator/model/apigatewayv2.py

```diff
@@ -57,39 +59,156 @@ class ApiGatewayV2ApiMapping(Resource):
 
 class ApiGatewayV2Authorizer(object):
     def __init__(
-        self, api_logical_id=None, name=None, authorization_scopes=[], jwt_configuration={}, id_source=None,
+        self,
+        api_logical_id=None,
+        name=None,
+        authorization_scopes=[],
+        jwt_configuration={},
+        id_source=None,
+        function_arn=None,
+        function_invoke_role=None,
+        identity=None,
+        authorizer_payload_format_version=None,
+        enable_simple_responses=None,
     ):
         """
         Creates an authorizer for use in V2 Http Apis
         """
-        if authorization_scopes is not None and not isinstance(authorization_scopes, list):
-            raise InvalidResourceException(api_logical_id, "AuthorizationScopes must be a list.")
-
-        # Currently only one type of auth
-        self.auth_type = "oauth2"
-
         self.api_logical_id = api_logical_id
         self.name = name
         self.authorization_scopes = authorization_scopes
-
-        # Validate necessary parameters exist
-        if not jwt_configuration:
-            raise InvalidResourceException(api_logical_id, name + " Authorizer must define 'JwtConfiguration'.")
         self.jwt_configuration = jwt_configuration
-        if not id_source:
-            raise InvalidResourceException(api_logical_id, name + " Authorizer must define 'IdentitySource'.")
         self.id_source = id_source
+        self.function_arn = function_arn
+        self.function_invoke_role = function_invoke_role
+        self.identity = identity
+        self.authorizer_payload_format_version = authorizer_payload_format_version
+        self.enable_simple_responses = enable_simple_responses
+
+        # Validate parameters
+        if authorization_scopes is not None and not isinstance(authorization_scopes, list):
+            raise InvalidResourceException(api_logical_id, "AuthorizationScopes must be a list.")
+
+        # Validate necessary parameters exist
```

Comment:
We also should validate that invalid parameters are not set (set to `None`) - so any parameters that are valid for only one auth type should be cause for rejection and an error if set for another auth type.

Reply:
Added more validation + updated tests.